runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/postprocessing/ProblemDefaultsProvider.java

```diff
@@ -26,7 +27,14 @@ public HttpProblem apply(HttpProblem problem, ProblemContext context) {
     }
 
     private URI defaultInstance(ProblemContext context) {
-        return context.path == null ? null : URI.create(context.path);
+        if (context.path == null) {
+            return null;
+        }
+        try {
+            return new URI(context.path.replaceAll(" ", "%20"));
```

Comment:
Question out of curiosity: shouldn't we escape whole path?

Reply:
As usual I'm referring to standards ;)

In [RFC2396](https://www.ietf.org/rfc/rfc2396.txt) (URI spec) you can find `unwise` characters that required escaping. 

```
  unwise      = "{" | "}" | "|" | "\" | "^" | "[" | "]" | "`"
```

Might not be worth doing it in this scenario though.

Reply:
Thanks for details @pazkooda, very good idea to properly handle all of them (i.e by using UrlDecoder/Encoder classes), please re-review, unfortunately the changes got bigger as I needed to adjust problem-to-json serializers as well.